halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/services/v1/ci/CiService.java

```diff
@@ -18,53 +18,53 @@
 
 import com.netflix.spinnaker.halyard.config.error.v1.ConfigNotFoundException;
 import com.netflix.spinnaker.halyard.config.error.v1.IllegalConfigException;
-import com.netflix.spinnaker.halyard.config.model.v1.node.Ci;
 import com.netflix.spinnaker.halyard.config.model.v1.node.Master;
+import com.netflix.spinnaker.halyard.config.model.v1.node.Ci;
 import com.netflix.spinnaker.halyard.config.model.v1.node.NodeFilter;
 import com.netflix.spinnaker.halyard.config.problem.v1.ConfigProblemBuilder;
 import com.netflix.spinnaker.halyard.config.services.v1.LookupService;
 import com.netflix.spinnaker.halyard.config.services.v1.ValidateService;
 import com.netflix.spinnaker.halyard.core.error.v1.HalException;
 import com.netflix.spinnaker.halyard.core.problem.v1.Problem.Severity;
 import com.netflix.spinnaker.halyard.core.problem.v1.ProblemSet;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Component;
+import lombok.RequiredArgsConstructor;
 
 import java.util.List;
 
 /**
  * This service is meant to be autowired into any service or controller that needs to inspect the current halconfigs
  * cis.
  */
-@Component
-public class CiService {
-  @Autowired
-  private LookupService lookupService;
+@RequiredArgsConstructor
```

Comment:
If all the sub-classes are meant to be spring beans, I think you can leave the `@Component` here.

Reply:
I just tried this now, and it didn't seem like Spring was creating beans for the subclasses.

Reply:
I stand corrected, must have misremembered that.